programs/ssl/ssl_server2.c

```diff
@@ -1629,36 +1629,67 @@ int main( int argc, char *argv[] )
     mbedtls_memory_buffer_alloc_init( alloc_buf, sizeof(alloc_buf) );
 #endif
 
+    ssl = mbedtls_calloc( 1, sizeof( *ssl ) );
+    conf = mbedtls_calloc( 1, sizeof( *conf ) );
+    cacert = mbedtls_calloc( 1, sizeof( *cacert ) );
+    srvcert = mbedtls_calloc( 1, sizeof( *srvcert ) );
+    pkey = mbedtls_calloc( 1, sizeof( *pkey ) );
+    srvcert2 = mbedtls_calloc( 1, sizeof( *srvcert2 ) );
+    pkey2 = mbedtls_calloc( 1, sizeof( *pkey2 ) );
+    cache = mbedtls_calloc( 1, sizeof( *cache ) );
+    timer = mbedtls_calloc( 1, sizeof( *timer ) );
+    ticket_ctx = mbedtls_calloc( 1, sizeof( *ticket_ctx ) );
+    entropy = mbedtls_calloc( 1, sizeof( *entropy ) );
+#if defined(MBEDTLS_CTR_DRBG_C)
+    ctr_drbg = mbedtls_calloc( 1, sizeof( *ctr_drbg ) );
+#else
+    hmac_drbg = mbedtls_calloc( 1, sizeof( *hmac_drbg ) );
+#endif
+
+    if( ssl       == NULL || conf      == NULL ||
+        cacert    == NULL || srvcert   == NULL ||
+        pkey      == NULL || srvcert2  == NULL ||
+        pkey2     == NULL || cache     == NULL ||
+#if defined(MBEDTLS_CTR_DRBG_C)
+        ctr_drbg  == NULL ||
+#else
+        hmac_drbg == NULL ||
+#endif
+        entropy   == NULL )
+    {
+        goto exit;
+    }
+
     /*
      * Make sure memory references are valid in case we exit early.
      */
     mbedtls_net_init( &client_fd );
     mbedtls_net_init( &listen_fd );
-    mbedtls_ssl_init( &ssl );
-    mbedtls_ssl_config_init( &conf );
+    mbedtls_ssl_init( ssl );
+    mbedtls_ssl_config_init( conf );
 #if defined(MBEDTLS_CTR_DRBG_C)
-    mbedtls_ctr_drbg_init( &ctr_drbg );
+    mbedtls_ctr_drbg_init( ctr_drbg );
 #else
-    mbedtls_hmac_drbg_init( &hmac_drbg );
+    mbedtls_hmac_drbg_init( hmac_drbg );
 #endif /* MBEDTLS_CTR_DRBG_C */
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
-    mbedtls_x509_crt_init( &cacert );
-    mbedtls_x509_crt_init( &srvcert );
-    mbedtls_pk_init( &pkey );
-    mbedtls_x509_crt_init( &srvcert2 );
-    mbedtls_pk_init( &pkey2 );
+    mbedtls_x509_crt_init( cacert );
+    mbedtls_x509_crt_init( srvcert );
+    mbedtls_pk_init( pkey );
+    mbedtls_x509_crt_init( srvcert2 );
+    mbedtls_pk_init( pkey2 );
 #if defined(MBEDTLS_SSL_ASYNC_PRIVATE)
-    memset( &ssl_async_keys, 0, sizeof( ssl_async_keys ) );
+    memset( ssl_async_keys, 0, sizeof( ssl_async_keys ) );
```

Comment:
The CI correctly points out that you didn't change the type of this variable, so you still need to take its address here (unless you prefer changing its type and allocation). (Multiple occurences.)

Reply:
Fixed.